torch/distributed/distributed_c10d.py

```diff
@@ -3234,3 +3235,41 @@ def new_subgroups_by_enumeration(
                 logger.info("Rank {} is assigned to subgroup {}".format(rank, ranks))
 
     return cur_subgroup, subgroups
+
+def broadcast_coalesced(
```

Comment:
Since this introduces a new public API, I wonder if we should have some discussion around expanding API surface of c10d library.

cc: @mrshenli @cbalioglu @zhaojuanmao for thoughts